### PR TITLE
Implemented publishing of Linea JVM libs by Gradle

### PR DESCRIPTION
**Pull request by Filter94**
This PR implements issue(s) #344

### Checklist

* [ ] I wrote new tests for my new core changes.
* [ ] I have successfully ran tests, style checker and build against my new changes locally.
* [ ] I have informed the team of any breaking changes if there are any.

### CODE REVIEW

File: jvm-libs/build.gradle [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  dependencies {
+    classpath(libs.jreleaser)
+  }
+}
+
+plugins {
+  id 'net.consensys.zkevm.kotlin-library-minimal-conventions'
+}
+
+Set<String> externalLibs = [
```
Comment (jpnovais) [inline jvm-libs/build.gradle]: suggestion: `publicApis`, I suggest to add a comment to what is the intention of these.

File: jvm-libs/build.gradle [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  "blob-decompressor",
+  "blob-shnarf-calculator",
+  "l1-rollup",
+  "l2-message-service"
+]
+
+subprojects {
+  // Note that artifacts with generated code only will be skipped if :publish is called without preceding :build
+  def hasGeneratedSources = it.fileTree(dir: 'build/generated', includes: ['**/*.java', '**/*.kt']).files.size() > 0
+  def hasSourceFiles = it.fileTree(dir: 'src', includes: ['**/*.java', '**/*.kt']).files.size() > 0
```
Comment (jpnovais) [inline jvm-libs/build.gradle:23]: the `subprojects {` with scans to the file system that run in the configuration phase can make gradle slower, did you measure the impact of this?

Even to run simple commands like `spotless, tasks, etc` we will pay the price... 

Also, I think checking just for `src` folder is good enough and may be lighter.

**Review by jpnovais — DISMISSED** [review]

File: jvm-libs/build.gradle (hunk 2 above)
Comment (Filter94) [reply jvm-libs/build.gradle:23]: > did you measure the impact of this?

I tested it as is and the way you suggest, it doesn't seem to have a noticeable impact

File: jvm-libs/build.gradle (hunk 1 above)
Comment (Filter94) [reply jvm-libs/build.gradle]: I'll change it to `publicApiLibs` and will add a comment

File: jvm-libs/build.gradle (hunk 2 above)
Comment (jpnovais) [reply jvm-libs/build.gradle:23]: Can you define noticeable? 
Unless gradle eagerly loads the file tree regardless, a deep scanning of all project's files must be slower...  Also, not everyone works on top of the range laptop with fast SSDs like us.

Comment (jpnovais) [reply jvm-libs/build.gradle:23]: To be precise, if takes more than 500ms IMO is worthy the optimization.
Do as you deam best

File: jvm-libs/linea/testing/teku-helper/build.gradle
Comment (jpnovais) [inline jvm-libs/linea/testing/teku-helper/build.gradle:1]: FYI, I removed this in MY WIP PR

**Review by jpnovais — APPROVED** [review]

File: jvm-libs/build.gradle (hunk 2 above)
Comment (Filter94) [reply jvm-libs/build.gradle:23]: > Can you define noticeable?

If there is any difference it's not noticeable with manual runs. In order to see the real difference I'd have to benchmark it with hundreds of runs and look at the average to measure the impact of this change

**Pull request merged**